gas/config/tc-riscv.c

```diff
@@ -1582,6 +1586,7 @@ validate_riscv_insn (const struct riscv_opcode *opc, int length)
 	case 'Z': /* RS1, CSR number.  */
 	case 'S': /* RS1, floating point.  */
 	case 's': USE_BITS (OP_MASK_RS1, OP_SH_RS1); break;
+    case 'g': /* RS1 and RS2 are the same.  */
```

Comment:
Here seems got a format problem, additionally, I noticed that the binutils upstream had applied the Zilsd patch, it aslo defined the register-pair, could also help to update this whole repo